runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -241,19 +240,17 @@ local function on_line_impl(self, buf, line)
     -- Some injected languages may not have highlight queries.
     if not highlighter_query:query() then return end
 
-    if state.iter == nil then
-      state.iter = highlighter_query:query():iter_captures(root_node, self.bufnr, line, root_end_row + 1)
-    end
+    local iter = highlighter_query:query():iter_captures(root_node, self.bufnr, line, root_end_row + 1, leftcol, rightcol)
```

Comment:
As I said in chat before, this is a perf regression, as you now need to retraverse the tree from the root on each line . What we possibly could do is to ask TS to add a method to _skip_ a range in a cursor (i e don't descend into irrelevant subtrees).

Reply:
Oh sorry about that, let me remove it